hypothesis_2.ipynb

```diff
@@ -199,22 +199,33 @@
     "    \n",
     "for row in users_lat:\n",
     "    username = row['username']\n",
+    "    # AW: It's extremely unsafe to evaluate external inputs.  Instead,\n",
+    "    # the file format should be deserializable with json, pickle, or\n",
+    "    # joined with a custom separator (eg. \"en|fr\")\n",
     "    languages = ast.literal_eval(row['language'])\n",
     "    langs = []\n",
     "    levels = []\n",
     "    edit_keys = {}\n",
     "    \n",
     "    for lang in languages:\n",
+    "        # AW: Push this into a separate function that can be tested.\n",
+    "        # AW: This fails for explicit native level \"en-N\"\n",
     "        # split by digits with a preceding character ('-') and include it\n",
     "        lang_extract = re.split('-(\\d)', lang)\n",
     "        # get the language\n",
     "        lang = lang_extract[0]\n",
     "        # get language level if present, else, assume native level\n",
     "        level = lang_extract[1] if len(lang_extract) > 1 else 5 \n",
     "        # append\n",
+    "        # AW: It seems fragile to pass the level through separately in\n",
```

Comment:
@adamwight 
[Solved] I see the point here. I am now using tuples and eventually creating two columns from them.